File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    std::string longestCommonPrefix(std::vector<std::string>& strs) {
        std::string res = "";
        for (int i = 0; i < strs[0].length(); ++i) {
            for (const auto& str: strs) {
                if (i == str.length() || str[i] != strs[0][i]) {
                    return res;
                }
            }
            res += strs[0][i];
        }
        return res;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String res = "";
        for (int i = 0; i < strs[0].length(); ++i) {
            for (String str: strs) {
                if (i == str.length() || str.charAt(i) != strs[0].charAt(i)) {
                    return res;
                }
            }
            res = res + strs[0].charAt(i);
        }
        return res;
    }
}
